include/circt/Dialect/FIRRTL/FIRRTLStructure.td

```diff
@@ -94,8 +95,10 @@ def FModuleOp : FIRRTLModuleLike<"module", [SingleBlock, NoTerminator]> {
     name, a list of ports, and a body that represents the connections within
     the module.
   }];
-  let arguments =
-            (ins DefaultValuedAttr<AnnotationArrayAttr, "{}">:$annotations);
+  let arguments = (ins
+    ConventionAttr:$convention,
+    DefaultValuedAttr<AnnotationArrayAttr, "{}">:$annotations
```

Comment:
Does declaring this with a default value of "internal" simplify some of the parsing logic later around checking for an empty attribute?

Reply:
As far as I can tell, `DefaultValuedAttr` doesn't help here, because we skip default builders and we have a custom asm printer/parser.